modules/web/js/ballerina/components/panel-decorator.jsx

```diff
@@ -57,14 +63,26 @@ class PanelDecorator extends React.Component {
     }
 
     onTitleInputBlur() {
-        this.setState({titleEditing: false})
+        this.setState({
+            titleEditing: false,
+            editingTitle: this.props.title
+        })
     }
 
     onTitleInputChange(e) {
-        const modelType = this.props.model.type.replace('Definition', '');
+        this.setState({editingTitle: e.target.value});
+    }
+
+    onTitleKeyDown(e) {
+        if(e.keyCode === 13) {
+            const modelType = this.props.model.type.replace('Definition', '');
 
-        // Setter functions take form 'setModelTypeName'. eg: setServiceName
-        this.props.model[`set${modelType}Name`](e.target.value);
+            // Setter functions take form 'setModelTypeName'. eg: setServiceName
+            this.props.model[`set${modelType}Name`](this.state.editingTitle);
```

Comment:
A bit risky to use a template to find and all a function. Can we pass the setter method as a prop?